ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte

```diff
@@ -4,7 +4,8 @@
 	export let value: Array<Array<string | number>> | string;
 	export let samples_dir: string;
 	let hovered = false;
-	let loaded = Array.isArray(value);
+	let loaded_value: Array<Array<string | number>> | string = value;
```

Comment:
The cause of the issue is that after click, the `value` is set back to the filename and since the file was already `loaded`, the filename was treated as an array in line 52.

Solution I went with is to store the file contents in a separate variable. That way we still only load the file once.